streaming/base/format/base/writer.py

```diff
@@ -15,32 +17,21 @@
 from streaming.base.compression import compress, get_compression_extension, is_compression
 from streaming.base.hashing import get_hash, is_hash
 from streaming.base.index import get_index_basename
+from streaming.base.storage.upload import CloudWriter
 
 __all__ = ['JointWriter', 'SplitWriter']
 
-
-def upload(local: str, remote: str) -> None:
-    """Placeholder upload method.
-
-    Args:
-        local (str): Local filename.
-        remote (str): Remote path.
-    """
-    dirname = os.path.dirname(remote)
-    if not os.path.exists(dirname):
-        os.makedirs(dirname)
-    import shutil
-    shutil.copy(local, remote)
+logger = logging.getLogger(__name__)
 
 
 class Writer(ABC):
     """Writes a streaming dataset.
 
     Args:
         local: (str, optional): Optional local output dataset directory. If not provided, a random
```

Comment:
In retrospect, I think the way we split the output argument into two (ie, `local` and `remote` as the inverse of `StreamingDataset`) is going to be too confusing for users.

What if instead we have one argument `out` which can be either local or remote. If it's local, we use that. If it's remote, we take a mkdtemp for our local. And we have a second argument nobody uses to override mkdtemp if they want.

Seems like that would be simpler? Thoughts?

Reply:
Based on the offline discussion, it is now changed to a single `out` argument which be either local or remote or list of [local, remote].